docs/example_nodejs.md

```diff
@@ -1,34 +1,313 @@
+
 ---
 title: "Example: NodeJS"
 layout: docs
 ---
 
-You're a NodeJS developer. You want to set up Tilt with your project.
+The best indicator of a healthy development workflow is a short feedback loop.
+
+Kubernetes is a huge wrench in the works.
+
+Let's fix this.
+
+In this example, we're going to take you through a simple “hello world” server written in NodeJS that uses [Mustache](https://mustache.github.io/) for templating and [Express](https://expressjs.com/) for serving.
+
+We'll use Tilt to:
+
+- Run the server on Kubernetes
+- Measure the time from a code change to a new process
+- Optimize that time for fast feedback
+
+Obviously, this particular server doesn't do much, but it's still a useful example to confirm that Tilt is working
+as expected in your environment.
+
+All the code is in this repo:
+
+[tilt-example-nodejs](https://github.com/windmilleng/tilt-example-nodejs)
+
+To skip straight to the fully optimized setup, go to this subdirectory:
 
-We hear you! We're working on a better guide for this.
+[Recommended Tiltfile](https://github.com/windmilleng/tilt-example-nodejs/tree/master/3-recommended)
 
-For now, [this GitHub issue](https://github.com/windmilleng/tilt/issues/2703) has some sample code.
+## Step 0: The Simplest Deployment
 
-For now, check out:
+Our server is just a few lines long, and all it does is serve us an HTML page:
 
-- [abc123](https://github.com/windmilleng/abc123) a mini microservice app with a simple HTTP NodeJS server called `letters`.
-- The [NodeJS microservice with Hot Reloading](nodejs_microservice_hotreloading.html) tutorial for using `live_update`.
-- Servantes (below), our microservice app in multiple languages.
+```javascript
+const express = require('express');
+const app = express();
+const path = require('path');
 
+app.use(express.static('public'));
 
-## Servantes
+app.get('/', function(req, res) {
+    res.sendFile(path.join(__dirname + '/index.html'));
+});
 
-[Servantes](https://github.com/windmilleng/servantes) is a personalized homepage
-app implemented with many microservices in multiple languages. You can run it
-with the commands
+app.listen(8000, () => {
+    console.log('Server running at http://localhost:8000/');
+});
```

Comment:
I find it a bit confusing that in both `listen()` and `get()` we're passing a function as the second argument, but one uses the regular function syntax, and the other uses the arrow function syntax. While technically correct, to me it feels like someone using Shakespearean English and Brooklyn English in the same sentence.

I'm not a JS developer though, so maybe I'm missing something.

Reply:
oh, there's no good reason from this, I was just stealing them from diff guides. will fix.